bytecomp/bytegen.ml

```diff
@@ -322,28 +322,43 @@ let add_pseudo_event loc modname c =
 
 (**** Compilation of a lambda expression ****)
 
-let try_blocks = ref []  (* list of stack size for each nested try block *)
+type stack_info = {
+  try_blocks : int list;
+  (* list of stack size for each nested try block *)
+  sz_static_raises : (int * (int * int * int list)) list;
+  (* association staticraise numbers -> (lbl,size of stack, try_blocks *)
+  max_stack_used : int ref;
```

Comment:
I always find the combination of immutable and mutable data in the same data structure confusing, but I believe the current use is correct, because all immutable copies of the data structure share the same `ref`. Getting rid of the ref would require `check_stack` to return a new `stack_info` and this would need propagating the new `stack_info` everywhere, which would make the change much more invasive. So I am inclined to accept the `ref` in the current version :)

Reply:
> and this would need propagating the new stack_info everywhere

In particular, the stack_info would have to be propagated up, where it's currently only propagated down.